src/neuroconv/tools/nwb_helpers/_models/_base_models.py

```diff
@@ -290,3 +290,15 @@ def __str__(self) -> str:
             string += f"\n{dataset_configuration}"
 
         return string
+
+    @classmethod
+    def from_nwbfile(cls, nwbfile: NWBFile) -> "BackendConfiguration":
+        from .._dataset_configuration import get_default_dataset_io_configurations
```

Comment:
Seems to be necessary for resolving what would otherwise be a circular dependency, mainly due to needing the other class types for proper annotation typing